### DISCUSSION TITLE
hasPreviousPage function in hibernate-reactive-panache PanacheQueryImpl doesn't return an Uni?

### DISCUSSION THREAD

**Discussion by beiertu-mms** [quarkusio/quarkus#38625] [Q&A] [answered]
Were there any reasons, on why the [`hasPreviousPage()`](https://github.com/quarkusio/quarkus/blob/3.7.1/extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/runtime/PanacheQueryImpl.java#L83) in the [hibernate-reactive-panache](https://github.com/quarkusio/quarkus/tree/3.7.1/extensions/panache/hibernate-reactive-panache) extension doesn't return a Uni like the other functions (`hasNextPage()`, `pageCount()` or `count()`) do?

[a comment by an automated account is not shown]

**Comment by DavideD** [comment, marked as answer]
I think the reason is that it doesn't need to be reactive.

The others are reactive because they need a query to get the result.
In the case of `.hasPreviousPage`, it just checks if the current page index is bigger than 0 (I think).

In term of API style, it's a bit weird though.

**Reply by beiertu-mms** [reply]
I see :)

> In term of API style, it's a bit weird though.

yes, that's true
